File: python/star-wars.py
#!/usr/bin/python3
import requests
import json

auth_token='sdfghjkloerdtfyguhiopfghjkl;fghjkl'
headers = {'Authorization': 'Bearer ' + auth_token}
data = {'app' : 'aaaaa'}

next = 'https://swapi.dev/api/people'

print('People...')

while next and len(next) > 0:
    response = requests.get(next, headers=headers)
    print('ok={} status={}'.format(response.ok, response.status_code))
    data = json.loads(response.text)
    next = data['next']
    print('results={} next={}'.format(len(data['results']), next))

print('People...Finished')

File: javascript/star-wars.js
import fetch from "node-fetch"

const getData = async (next) => {
    const response = await fetch(next)
    const data = await response.json()
    return data
}

var next = 'https://swapi.dev/api/people'

console.log("People...")

while (next && next.length > 0) {
    var data = await getData(next)
    console.log(data.results.length +" - "+ data.next)
    next = data.next
}

console.log("People...Finished")
